src/core/core.adapters.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @namespace Chart._adapters
+ * @since 2.8.0
+ * @private
+ */
+
+import type {AnyObject} from '../../types/basic';
+import type {BindContextToMethods} from '../../types/utils';
+import type {ChartOptions} from '../../types';
+
+export type TimeUnit = 'millisecond' | 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'quarter' | 'year';
+
+function abstract<T = void>(): T {
+  throw new Error('This method is not implemented: Check that a complete date adapter is provided.');
+}
+
+/**
+ * Date adapter (current used by the time scale)
+ * @namespace Chart._adapters._date
+ * @memberof Chart._adapters
+ * @private
+ */
+export class DateAdapter<T extends AnyObject = AnyObject> {
+
+  /**
+   * Override default date adapter methods.
+   * Accepts type parameter to define options type.
+   * @example
+   * Chart._adapters._date.override<{myAdapterOption: string}>({
+   *   init() {
+   *     console.log(this.options.myAdapterOption);
+   *   }
+   * })
+   */
+  static override<T extends AnyObject = AnyObject>(
+    members: Partial<BindContextToMethods<Omit<DateAdapter, 'options'>, DateAdapter<T>>>
+  ) {
+    Object.assign(DateAdapter.prototype, members);
+  }
+
+  readonly options: T;
+
+  constructor(options: T) {
+    this.options = options || {} as T;
+  }
+
+  /**
+   * Will called with chart options after adapter creation.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  init(chartOptions: ChartOptions) {}
+
+  /**
+	 * Returns a map of time formats for the supported formatting units defined
+	 * in Unit as well as 'datetime' representing a detailed date/time string.
+	 */
+  formats(): Record<string, string> {
+    return abstract();
+  }
+
+  /**
+	 * Parses the given `value` and return the associated timestamp.
+	 * @param value - the value to parse (usually comes from the data)
+	 * @param [format] - the expected data format
+	 */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
it's pretty ugly having to put this at every method. `tsc` can alert to unused variables. in another project of mine we recently switched from eslint to prettier + `tsc` and so far it's been a nicer combination. I wonder if we should consider that

Reply:
@benmccann eslint, besides style rules, contains rules to escape logical errors in code

Reply:
any that `tsc` won't catch though?

Reply:
> I wonder if we should consider that

Anyway, we can discuss it, but separately from this PR.

Reply:
yeah, agree it would be a separate change. but all these disables do make me wonder if we're doing this in the most typescript-friendly way. perhaps we should remove all the parameters from this class and declare those in an `interface` instead?

Reply:
@benmccann done